pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2567,7 +2570,16 @@ type Counter struct {
 	Value string `json:"value" protobuf:"bytes,1,opt,name=value"`
 }
 
-// Memoization enables caching for the Outputs of the template
+// Memoization specifies the storage for cache that can be used for all templates at workflow level
+type Memoization struct {
```

Comment:
Why do we need a new type here? What's wrong with using `Memozie`?

Reply:
`Memoize.Key` should be set at template level. Perhaps we can reuse `Memoize` and then validate whether the workflow-spec memoize contains `Key` in `executeTemplate`?

Reply:
Yes, that sounds much better.